app/helpers/application_helper.rb

```diff
@@ -1,2 +1,24 @@
 module ApplicationHelper
-end
+  def find_type(url)
+    api_key = Rails.application.secrets.embedly_api_key
+    embedly_api = Embedly::API.new(key: api_key)
+    obj = embedly_api.oembed :url => url
+    obj.first.type
+  end
+  def embed_media(url, size)
+    api_key = Rails.application.secrets.embedly_api_key
+    embedly_api = Embedly::API.new(key: api_key)
+    obj = embedly_api.oembed :url => url, :width => size
+    if obj.first.type == 'video'
+      raw(obj.first.html)
+    else
+      raw(obj.first.thumbnail_url) 
+    end
+  end
+  def embed_description(url)
+    api_key = Rails.application.secrets.embedly_api_key
+    embedly_api = Embedly::API.new(key: api_key)
+    obj = embedly_api.oembed :url => url
```

Comment:
Each `embedly_api.oembed` call will make an HTTP request to Embedly's servers. Calls to external services like this should be minimized or avoided for performance.

Update the code in the helper and in `app/views/bookmarks/show.html.erb` so that the embedly API is only called once .